### PR TITLE
pylintrc: re-enable broad-exception-raised

### PR DESCRIPTION
**Pull request by mustafakemalgilor**
replace all Exception() with appropriate exception types. Used built-in exception types where possible, added user-defined exception types to hotsos.core.exceptions for others.

### CODE REVIEW

File: hotsos/core/config.py [hunk 1]
```diff
@@ -177,7 +178,8 @@ def __setitem__(self, key, item):
                 item = group.opts[key].value_type(item)
                 break
         else:
-            raise Exception(f"config option '{key}' not found in any optgrpup")
+            raise NameError(
+                f"config option '{key}' not found in any optgrpup")
```
Comment (brianphaley) [inline hotsos/core/config.py:182]: There is a typo here, 'optgrpup', but it was here before.

File: hotsos/core/ycheck/common.py [hunk 2]
```diff
@@ -261,9 +262,10 @@ def validate(self, plugin_name):
         search belongs.
         """
         if self.plugin_name is None:
-            raise Exception(f"{self.__class__.__name__}.plugin_name must set "
-                            "to the name of the plugin in which it will run "
-                            f"(currently={self.plugin_name})")
+            raise NameNotSetError(
+                f"{self.__class__.__name__}.plugin_name must set "
```
Comment (brianphaley) [inline hotsos/core/ycheck/common.py:266]: nit: "must be set" - if you change the other typo

**Review by brianphaley — APPROVED** [review]

File: tests/unit/utils.py [hunk 3]
```diff
@@ -130,8 +136,9 @@ def check_raised_issues(test_inst, expected, actual):
             test_inst.assertNotIn('potential-issues', actual)
             return
         if 'potential-issues' not in actual:
-            raise Exception("test expects one or more issues to have "
-                            "been raised did not find any.")
+            raise ExpectationNotMetError(
+                "test expects one or more issues to have "
+                "been raised did not find any.")
```
Comment (pponnuvel) [inline tests/unit/utils.py:141]: "..**but** did not..."

File: tests/unit/utils.py [hunk 4]
```diff
@@ -55,9 +60,9 @@ def _inner(cls):
         for testdef in find_all_templated_tests(_path):
             tg = TemplatedTestGenerator(path, testdef)
             if hasattr(cls, tg.test_method_name):
-                raise Exception("test name conflict for "
-                                f"'{tg.test_method_name}' - "
-                                "a test with this name already exists")
+                raise NameAlreadyRegisteredError(
+                    f"test name conflict for '{tg.test_method_name}' - "
```
Comment (pponnuvel) [inline tests/unit/utils.py:64]: Maybe saying it once is enough? 
```
raise NameAlreadyRegisteredError(
                    f"a test with the name '{tg.test_method_name}' already exists")
```

**Review by pponnuvel — APPROVED** [review]

File: hotsos/core/config.py (hunk 1 above)
Comment (mustafakemalgilor) [reply hotsos/core/config.py]: done

File: hotsos/core/ycheck/common.py (hunk 2 above)
Comment (mustafakemalgilor) [reply hotsos/core/ycheck/common.py]: done

File: tests/unit/utils.py (hunk 3 above)
Comment (mustafakemalgilor) [reply tests/unit/utils.py:146]: done

File: tests/unit/utils.py (hunk 4 above)
Comment (mustafakemalgilor) [reply tests/unit/utils.py]: done

File: hotsos/core/config.py [hunk 5]
```diff
@@ -177,7 +178,8 @@ def __setitem__(self, key, item):
                 item = group.opts[key].value_type(item)
                 break
         else:
-            raise Exception(f"config option '{key}' not found in any optgrpup")
+            raise NameError(
```
[the remark this answers is not in the document]
Comment (mustafakemalgilor) [reply hotsos/core/config.py]: done

**Pull request merged**